keras_retinanet/models/retinanet.py

```diff
@@ -193,28 +311,71 @@ def retinanet(
     features = create_pyramid_features(C3, C4, C5)
 
     # for all pyramid levels, run available submodels
-    pyramid = __build_pyramid(submodels, features)
-    anchors = __build_anchors(anchor_parameters, features)
+    pyramids = __build_pyramid(submodels, features)
+    anchors   = __build_anchors(anchor_parameters, features)
+
+    # concatenate outputs to one list
+    outputs = [anchors] + pyramids
+    if output_fpn:
+        outputs += features
 
-    return keras.models.Model(inputs=inputs, outputs=[anchors] + pyramid, name=name)
+    return keras.models.Model(inputs=inputs, outputs=outputs, name=name)
 
 
-def retinanet_bbox(inputs, num_classes, nms=True, name='retinanet-bbox', *args, **kwargs):
-    model = retinanet(inputs=inputs, num_classes=num_classes, *args, **kwargs)
+def retinanet_bbox(
+    inputs,
+    num_classes,
+    nms=True,
+    name='retinanet-bbox',
+    output_fpn=False,
+    **kwargs
+):
+    """ Construct a RetinaNet model on top of a backbone and adds convenience functions to output detections directly.
+
+    This model uses the minimum retinanet model and appends a few layers to compute detections within the graph.
+    These layers include applying the regression values to the anchors and performing NMS.
+
+    Args
+        inputs      : keras.layers.Input (or list of) for the input to the model.
+        num_classes : Number of classes to classify.
+        output_fpn  : If True, the last 5 outputs before detections are P3...P7.
+        name        : Name of the model.
+        *kwargs     : Additional kwargs to pass to the minimal retinanet model.
+
+    Returns
+        Model with inputs as input and as output the output of each submodel for each pyramid level and the detections.
+
+        The order is as defined in submodels. Using default values the output is:
+        ```
+        [
+            regression, classification, detections
+        ]
+        ```
+    """
+    model = retinanet(inputs=inputs, num_classes=num_classes, output_fpn=output_fpn, **kwargs)
 
     # we expect the anchors, regression and classification values as first output
     anchors        = model.outputs[0]
     regression     = model.outputs[1]
     classification = model.outputs[2]
 
+    if output_fpn:
+        other = model.outputs[3:-5]
```

Comment:
Perhaps more clarification on what other constitutes? Clearly relates to having output_fpn set to True. Maybe a comment just above L362?

Reply:
:+1: